scripts/ldmx-compile.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+############################# Usage #############################
+# ldmx ./scripts/ldmx-compile.sh
+# for changing the defaults
+# ldmx setenv LDMX_COMPILE_CORES=<num cores to be used>, i.e.
+# ldmx setenv LDMX_COMPILE_CORES=16
+# or
+# ldmx setenv LDMX_COMPILE_BUILD=<build location>
+
+if [ -z "$LDMX_COMPILE_CORES" ]; then
+    LDMX_COMPILE_CORES=$(nproc)
```

Comment:
I'm not super committed to this, but maybe we should have the default be one core. I say this because https://github.com/LDMX-Software/Tracking/issues/71 but I'm not sure how isolated that issue is.

Reply:
I'd prefer to keep it on `$(nproc)`, the issue you pointed to seems isolated and it takes quite some time to compile on a single core. If somebody runs into issues with the multi-core compilation they can still recompile with a single core, but given that is not what happens most of the time, I'd say let's keep the default on `$(nproc)`.

Reply:
Sounds reasonable to me :+1: